src/main/java/kr/codesqaud/cafe/service/ArticleService.java

```diff
@@ -63,6 +71,15 @@ public void editArticle(final Long articleId, final ArticleEditRequest request)
 	public void deleteArticle(final Long articleId) {
 		articleRepository.findById(articleId)
 			.orElseThrow(() -> new NotFoundException(String.format("%d번 게시글을 찾을 수 없습니다.", articleId)));
-		articleRepository.deleteById(articleId);
+		articleRepository.isPossibleDeleteById(articleId)
+			.ifPresentOrElse(isExists -> {
+				if (isExists) {
+					articleRepository.deleteById(articleId);
+					return;
+				}
+				throw new InvalidOperationException(articleId);
+			}, () -> {
+				throw new InvalidOperationException(articleId);
+			});
```

Comment:
```suggestion
 if (isPossibleToDelete(articleId)) {
        articleRepository.deleteById(articleId);
 } 
 throw new InvalidOperationException(articleId);
```
위와 같은 형태로 개선할 수는 없을까요?ㅎㅎ

Reply:
넵! 수정했습니다! 😃